[GEOT-5851] Fixed line gap functionality for MBStyle line layers

johngschulz opened the pull request.

bradh commented on modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/layer/LineMBLayer.java:
```diff
@@ -533,23 +533,17 @@ public boolean hasLinePattern() {
-                    sf.description(Text.text("line"), null), NonSI.PIXEL, blankStroke, lineOffset());
-            symbolizers.add(gapOuterLine);
-            symbolizers.add(blankLine);
+        	    Double topOffset = getLineOffset().doubleValue() + (getLineGapWidth().doubleValue() / 2d) + getLineWidth().doubleValue()/2d ;
```
Possible issue with indentation here?

johngschulz commented:
I addressed the indentation issue, and updated the commit to include GEOT-5851 in the title, as it appears to be a duplicate of GEOS-8346 and this should fix both.

bradh commented:
Travis is failing to build the MBStyles module. You might want to take a look at that.

johngschulz commented:
I've had some issues with timing out on the build locally - but it looks like I forgot to update the visual transformation tests to reflect the new changes.

tbarsballe commented:
On the GEOS vs GEOT issue, I had incorrectly reported this issue against GEOS instead of GEOT.
I moved the ticket over on Friday, so any confusion there is my fault.

tbarsballe commented on modules/unsupported/mbstyle/src/test/java/org/geotools/mbstyle/transform/StyleTransformTest.java:
```diff
@@ -256,6 +256,16 @@ public void testLineGapWidth() throws IOException, ParseException {
+		} catch (TransformerException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
```
Looks like tabs are being used here - should be spaces.

johngschulz replied:
I actually meant to remove that before pushing - I was just using that to debug/test the sld output.

tbarsballe commented:
Looks good. One last request - please include the JIRA ticket number in your commit message. It makes navigating the git history easier.

johngschulz commented:
Updated commit message with Jira ticket number.

tbarsballe commented:
Backported to 18.x
